src/main/java/com/code_intelligence/jazzer/runtime/TraceCmpHooks.java

```diff
@@ -69,6 +73,14 @@ final public class TraceCmpHooks {
     TraceDataFlowNativeCallbacks.traceCmpLong(thisObject, (long) arguments[0], hookId);
   }
 
+  @MethodHook(type = HookType.BEFORE, targetClassName = "clojure.lang.Numbers", targetMethod = "isZero")
+  @MethodHook(type = HookType.BEFORE, targetClassName = "clojure.lang.Numbers", targetMethod = "isPos")
+  @MethodHook(type = HookType.BEFORE, targetClassName = "clojure.lang.Numbers", targetMethod = "isNeg")
+  public static void staticLongCompareZero(
+          MethodHandle method, Object thisObject, Object[] arguments, int hookId) {
+    TraceDataFlowNativeCallbacks.traceCmpLong((long) arguments[0], (long) 0, hookId);
```

Comment:
Every hook should either use `targetMethodDescriptor` to specify the exact function signature it matches *or* check the types of arguments before casting. Doing neither can result in `ClassCastException`s if unexpected (or new) overloads are invoked.

Reply:
They aren't overloaded *yet* - if new versions of Clojure are released, this might change and we don't want code to break until Jazzer is updated. Adding a `targetMethodDescriptor` ensures that this won't happen and has no runtime cost.

That said, if the signature really is `(java.lang.Number)Z`, then the casts to `(long)` aren't correct if the argument is e.g. an `int`. You can cast to `Number` instead and use `longValue`. 

Also please add a test that exercises these hooks with different types. Instead of compiling Clojure, which may be tedious to set up, you can add the clojure stdlib as a Maven dep and just call the Java functions directly.

Reply:
Ah, yes, thanks for the hint!

I've now moved these hooks into their own functions concerning `java.lang.Number` objects. A test is added too.